lib/rogue.js

```diff
@@ -19,27 +20,70 @@ function executePost(endpoint, data) {
     .post(url)
     .set(defaultConfig.apiKeyHeader, defaultConfig.clientOptions.apiKey)
     .send(data)
-    .then(res => res.body)
-    .catch(err => err);
+    .then(res => res.body);
 }
 
 /**
  * @param {object} req
- * @return {Promise}
+ * @return {object}
  */
-module.exports.postSignupForReq = function (req) {
+function getDefaultPayloadForReq(req) {
   const campaign = req.campaign;
   const userId = req.userId;
-
   // @see https://github.com/DoSomething/rogue/blob/master/documentation/endpoints/signups.md#signups
   const data = {
     source: defaultConfig.source,
-    // TODO: This should happen in Phoenix JS.
     campaign_id: Number(campaign.id),
     campaign_run_id: Number(campaign.currentCampaignRun.id),
     northstar_id: userId,
   };
-  logger.debug('rogue.postSignupForReq', { data });
+  return data;
+}
+
+/**
+ * @param {object} req
+ * @return {Promise}
+ */
+module.exports.createSignupForReq = function (req) {
+  const data = getDefaultPayloadForReq(req);
+  logger.debug('rogue.createSignupForReq', { data });
 
   return executePost('signups', data);
 };
+
+/**
+ * @param {string} url
+ * @return {Buffer}
+ */
+function fetchDataForImageUrl(url) {
+  logger.debug(`rogue.getDataForImageUrl:${url}`);
+
+  // @see https://github.com/visionmedia/superagent/issues/871#issuecomment-286199206
+  return superagent.get(url)
+    .buffer(true)
+    .parse(superagent.parse.image);
+}
+
+/**
+ * @param {object} req
+ * @return {Promise}
+ */
+module.exports.createPostForReq = function (req) {
+  const data = getDefaultPayloadForReq(req);
+  const draft = req.signup.draft_reportback_submission;
+
+  // @see https://github.com/DoSomething/rogue/blob/master/documentation/endpoints/posts.md
+  data.quantity = draft.quantity;
+  data.caption = draft.caption;
+  data.remote_addr = ip.address();
```

Comment:
Rogue throws an error if a [`remote_addr` param is not passed](https://rpm.newrelic.com/accounts/108038/applications/37465261/filterable_errors#/show/ceac6d95-ae3f-11e7-b130-0242ac110004_4887_10880/stack_trace?top_facet=transactionUiName&primary_facet=error.class&barchart=barchart&filters=%5B%7B%22key%22%3A%22error.class%22%2C%22value%22%3A%22ErrorException%22%2C%22like%22%3Afalse%7D%5D&_k=07zfst).